### ISSUE TITLE
[netrw] Netrw will ignore any files after a directory when using :bnext

### ISSUE THREAD

**Issue by joereynolds** [vim/vim#5073]
## The Bug
When I open vim with `vim *` and there's a directory present in the buffers, using `:bnext` will restart the cycle once it hits the directory.

### Example

If I have the following file list

```
├── file1.php
├── another-file.php
├── one-more.php
├── A-Directory
├── a-file-after-the-directory-in-the-list.php
└── and-another.php
```

When navigating through using `:bnext` (starting at `file1.php`), once it hits `A-Directory` it will begin cycling at `file1.php` instead of continuing on to `a-file-after-the-directory-in-the-list.php`

## Description
`:bnext` will restart at the 0th buffer once it hits a directory in the list in `:ls`

## Steps

1. Run `vim *` on a directory that has both files and a directory
2. Go through the buffers with `:bnext`
3. Once it hits the directory, the next `:bnext` will begin at the start of the list of buffers instead of continuing on to the next buffer.

## Expected

I would expect it to the go the next buffer in the list, not restart the cycle.

## Environment
 - Vim version 8.0.1453
 - OS: Ubuntu 18.04
 - Terminal: GNOME Terminal

## Additional
- This does not happen with `let loaded_netrwPlugin = 1` in the vimrc.
- I performed this on a clean installation of vim with nothing in my vimrc.
- Possibly related to #2597

**Comment by chrisbra** [comment]
Yes, what happens is. Vim notices that it tries to open a directory. Then netrw kicks in and creates a new unnamed buffer, that contains the filecontent for that buffer. As it created a new buffer, this buffer will be at then end of the current buffer list, so the next `:bnext` wraps over and starts from the beginning.

**Comment by chrisbra** [comment]
this looks indeed like a duplicate of #2597
